case-lib/pipeline.sh

```diff
@@ -2,6 +2,19 @@
 
 func_pipeline_export()
 {
+    # For legacy HDA platform, there is no topology, we have to exporting pipeline
+    # information from proc file system.
+    is_sof_used || {
+        filter_str="$@"["$#"]
+        dlogi "No SOF sound card found, exporting pipeline parameters from proc file system"
+        tmp_pipeline_params=$(mktemp /tmp/pipeline-params.XXXXXXXX)
+        sof-dump-status.py -e "$filter_str" > "$tmp_pipeline_params" ||
+            die "Failed to export pipeline parameters from proc file system"
+        source "$tmp_pipeline_params" || return 1
+        rm "$tmp_pipeline_params" || true
```

Comment:
`rm -f "$tmp_pipeline_params"` never fails if you ever need that but why would `rm "$tmp_pipeline_params"` fail anyway? If this simple `rm` fails then something must have gone terribly wrong and we probably want to let `set -e` stop there.